### PR TITLE
removed ES auth info from logs

### PR DESCRIPTION
**Pull request by gggopi**
Please review. Thanks

### CODE REVIEW

File: stagemonitor-core/src/main/java/org/stagemonitor/core/CorePlugin.java [hunk 1]
```diff
@@ -699,7 +691,7 @@ public String getElasticsearchUrl() {
 		return result;
 	}
 
-	private String removeUserInfo(String url) {
+	public String removeUserInfo(String url) {
```
Comment (felixbarny) [inline stagemonitor-core/src/main/java/org/stagemonitor/core/CorePlugin.java]: remove method here

File: stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java [hunk 2]
```diff
@@ -182,4 +181,14 @@ public Integer handleResponse(InputStream is, Integer statusCode, IOException e)
 			return statusCode;
 		}
 	}
+
+  	private String removeUserInfo(String url) {
```
Comment (felixbarny) [inline stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java]: make public static

File: stagemonitor-core/src/main/java/org/stagemonitor/core/elasticsearch/ElasticsearchClient.java [hunk 3]
```diff
@@ -222,7 +222,7 @@ public void updateIndexSettings(String indexPattern, Map<String, ?> settings) {
 			return;
 		}
 		final String url = corePlugin.getElasticsearchUrl() + "/" + indexPattern + "/_settings?ignore_unavailable=true";
-		logger.info("Updating index settings {}\n{}", url, settings);
+		logger.info("Updating index settings {}\n{}", corePlugin.removeUserInfo(url), settings);
```
Comment (felixbarny) [inline stagemonitor-core/src/main/java/org/stagemonitor/core/elasticsearch/ElasticsearchClient.java]: use HttpClient.removeUserInfo

File: stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java [hunk 4]
```diff
@@ -81,6 +79,7 @@ public int send(final String method, final String url, final Map<String, String>
 
 		HttpURLConnection connection = null;
 		InputStream inputStream = null;
+		String urlWithoutAuthInfo = removeUserInfo(url);
```
Comment (felixbarny) [inline stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java]: inline variable. if there are problems with the removeUserInfo method it limits the scope

File: stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java (hunk 2 above)
Comment (gggopi) [inline stagemonitor-core/src/main/java/org/stagemonitor/core/util/HttpClient.java]: I would have to make a separate static logger for this method then? 	  		
logger.warn("Suppressed exception", e);

**Comment by felixbarny** [comment]
Also please sign the [CLA](https://github.com/stagemonitor/stagemonitor/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla)

**Comment by gggopi** [comment]
test cases fixed - had to add a null check in the removeUserInfo.
signed cla

**Pull request merged**